### PR TITLE
[2.16.x] DDF-5126 Add doPrivilged for UrlResourceReader

### PR DESCRIPTION
**Pull request by tyler30clemens**
#### What does this PR do?
It was discovered that the URL Resource Reader was failing to work with resources outside of certain directories because it needed a doPrivileged block around the canonical path. This PR fixes that to allow it to work with other directories as well.

#### Who is reviewing it? 
@austinsteffes
@Kjames5269

#### Ask 2 committers to review/merge the PR and tag them here.
@ahoffer 
@brjeter
@stustison

#### How should this be tested?
(see me for test resources and finer details)
0. Create path outside of DDF and put image resource there

1. Unzip and edit configurations.policy file to have permissions to directory in step 0
2. Install standard
3. Edit URL Resource Reader configuration to have access to path in step 0
4. Ingest xml resource through command line (make sure resource-uri property is same as step 0)
5. Go to Intrigue and hit download button on ingested product
6. Verify no issues and image is returned

#### What are the relevant tickets?
Work for: #5126 

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by tyler30clemens** [comment]
build now

File: distribution/ddf-common/src/main/resources/security/configurations.policy [hunk 1]
```diff
@@ -84,7 +84,7 @@ grant codeBase "file:/catalog-core-directorymonitor/org.apache.camel.camel-core/
 //   directory being monitored.
 //
 // DO NOT MODIFY THE NEXT LINE. It specifies which  modules are granted permission.
-grant codeBase "file:/org.apache.tika.core/catalog-core-urlresourcereader" {
+grant codeBase "file:/org.apache.tika.core/catalog-core-urlresourcereader/catalog-core-standardframework" {
```
Comment (peterhuffer) [inline distribution/ddf-common/src/main/resources/security/configurations.policy]: ❓ Why is this needed all the sudden? We do a privileged call when the resource is fetched
https://github.com/codice/ddf/blob/master/catalog/core/catalog-core-urlresourcereader/src/main/java/ddf/catalog/resource/impl/URLResourceReader.java#L329

Comment (peterhuffer) [inline distribution/ddf-common/src/main/resources/security/configurations.policy]: ❓ Do we need to do a privileged call from somewhere in the UrlResourceReader instead?

Comment (tyler30clemens) [reply distribution/ddf-common/src/main/resources/security/configurations.policy]: This was the primary solution presented by the AC debugger and it seemed reasonable. Perhaps there is an alternate way by adding privileged calls instead, but this is a port of an already merged master PR.

Comment (tyler30clemens) [reply distribution/ddf-common/src/main/resources/security/configurations.policy]: After looking at it further, there is a better solution with a doPrivileged block. Updating this PR and adding another PR to fix this in the master branch.

**Comment by tyler30clemens** [comment]
build now
